src/methods/index.ts

```diff
@@ -114,14 +115,14 @@ export const methods = (storage: Storage): SignatoryMethodMapping => {
     sign: async (dataToSign: string, address: string, passphrase: string, chainId: types.ChainId) => {
       const wallet = await storage.getAccount(address);
       const acct = Wallet.fromV3(wallet.keystore, passphrase);
-      return personalSign(Buffer.from(dataToSign), acct.getPrivateKey(), parseInt(chainId, 16));
+      return personalSign(Buffer.from(hexToString(dataToSign)), acct.getPrivateKey(), hexToNumber(chainId));
```

Comment:
In this instance, here the spec. requires dataToSign to be already in hexString format. The open-rpc document adds this validation. The reason for hex all the way down is to not hide or push any conversion logic into the signer itself. One case might be that you have a string that isn't utf-8 encoded but in some other encoding scheme, we don't get into a game of figuring out how to deal with that we just assume hex. and move forward. The server itself does the validation. it's easiest to check this via the inspector. @ inspector.open-rpc.org.

Reply:
amazing! and i see what you meant. alright, updated the PR and also fixed sig verification on my end https://github.com/vbstreetz/signatory-client/commit/c0fff9b3afc3572fa03c8ae115bc109c0ad0120f#diff-0acd5d0fed8f072e1aa9ba4d4ea68f9aR21.